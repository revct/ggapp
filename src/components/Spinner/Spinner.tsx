import React from 'react';
import {
  ActivityIndicator,
  Platform,
  ActivityIndicatorProps,
} from 'react-native';
import colors from 'configs/colors';

export interface SpinnerProps extends ActivityIndicatorProps {
  size?: 'large' | 'small';
}

export const Spinner: React.FC<SpinnerProps> = function({
  color,
  size,
  ...props
}) {
  return (
    <ActivityIndicator
      {...props}
      color={Platform.select({
        android: color ? color : colors.accent,
        ios: color ? color : colors.gray700,
      })}
      size={size || 'small'}
    />
  );
};

export default Spinner;
