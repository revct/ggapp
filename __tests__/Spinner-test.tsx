import 'react-native';
import React from 'react';
import {act, create} from 'react-test-renderer';
import {Spinner} from '../src/components/Spinner/Spinner';

describe('Spinner', () => {
  describe('Rendering', () => {
    it('should match snapshot', () => {
      let component:any;
      act(() => {
        component = create(<Spinner />);
      });
      expect(component.toJSON()).toMatchSnapshot();
      act(() => {
        component.update(<Spinner size="large" />);
      });
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
