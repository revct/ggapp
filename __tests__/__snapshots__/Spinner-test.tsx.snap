// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Rendering should match snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="#616161"
  hidesWhenStopped={true}
  size="small"
/>
`;

exports[`Spinner Rendering should match snapshot 2`] = `
<ActivityIndicator
  animating={true}
  color="#616161"
  hidesWhenStopped={true}
  size="large"
/>
`;
